WordPress/src/main/java/org/wordpress/android/ui/posts/PostListType.kt

```diff
@@ -0,0 +1,32 @@
+package org.wordpress.android.ui.posts
+
+import org.wordpress.android.R.string
+import org.wordpress.android.fluxc.model.post.PostStatus
+import org.wordpress.android.fluxc.model.post.PostStatus.PRIVATE
+
+enum class PostListType(val postStatuses: List<PostStatus>) {
+    PUBLISHED(listOf(PostStatus.PUBLISHED, PostStatus.PRIVATE)),
+    DRAFTS(listOf(PostStatus.DRAFT, PostStatus.PENDING)),
+    SCHEDULED(listOf(PostStatus.SCHEDULED)),
+    TRASHED(listOf(PostStatus.TRASHED));
+
+    val titleResId: Int
+        get() = when (this) {
+            PUBLISHED -> string.post_list_published
+            DRAFTS -> string.post_list_drafts
+            SCHEDULED -> string.post_list_scheduled
+            TRASHED -> string.post_list_trashed
+        }
+
+    companion object {
+        fun fromPostStatus(status: PostStatus): PostListType {
+            return when (status) {
+                PostStatus.PUBLISHED, PRIVATE -> PUBLISHED
+                PostStatus.DRAFT, PostStatus.PENDING -> DRAFTS
+                PostStatus.TRASHED -> TRASHED
+                PostStatus.SCHEDULED -> SCHEDULED
+                PostStatus.UNKNOWN -> TODO()
```

Comment:
> > I'm not sure what to do with posts with status "Unknown"

> It seems we just simply filter "Unknown" Pages out (PostStore.getPagesFromDb()). Should we do the same for Posts?

Do we know when the post status will be `Unknown`? Knowing that could help us make a decision and document it.

Reply:
I haven't found anything useful yet. I checked the FluxC code and it's used when the `status` is empty or it isn't any of the known statuses (PUBLISHED, DRAFT, PRIVATE, PENDING, TRASHED, SCHEDULED). However, I'm wondering how come it can actually happen.

Reply:
I remembered from the Slack conversation today, I believe it's for self-hosted sites because the status names can be changed by them (or new ones added). Let's open a low priority issue for this for now and leave the todo.

Reply:
Opened #9252 for this.